lib/flatpak-installation.c

```diff
@@ -1176,16 +1167,25 @@ progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
   fetched = ostree_async_progress_get_uint (progress, "fetched");
   metadata_fetched = ostree_async_progress_get_uint (progress, "metadata-fetched");
   requested = ostree_async_progress_get_uint (progress, "requested");
+  total_bytes = total_extra_data_bytes + total_delta_part_size + requested;
+  total_transferred_bytes = transferred_extra_data_bytes + bytes_transferred + fetched;
   current_time = g_get_monotonic_time ();
 
+  /* When we receive the status, it means that the ostree pull operation is
+   * finished. We only have to be careful about the extra-data fields. */
   if (status && !downloading_extra_data)
     {
       g_string_append (buf, status);
 
-      /* The status is sent on error or when the pull is finished */
-      new_progress = outstanding_extra_data ? 55 : 100;
+      if (outstanding_extra_data == 0)
+        new_progress = 100;
+      else
+        new_progress = 10 + ((bytes_transferred + fetched) / (gdouble) total_bytes) * 90;
```

Comment:
I don't think this is right. We should not involve fetched and requested here, because they are object counts, not sizes. So they are not really comparable. Since in this case we only care about the extra data, bytes_transferred should be fine to use, as it is the total amount of bytes the pull operation ever downloaded.

So, this should probably be: 10 + (bytes_transferred / (bytes_transferred + total_extra_data_bytes) * 90

No?